pulp_service/pulp_service/app/authorization.py

```diff
@@ -0,0 +1,47 @@
+from contextvars import ContextVar
+from base64 import b64decode
+from binascii import Error as Base64DecodeError
+import json
+import jq
+from pulpcore.app import settings
+from pulp_service.app.models import DomainOrg
+from rest_framework.permissions import BasePermission
+
+
+org_id_var = ContextVar('org_id')
+
+class DomainBasedPermission(BasePermission):
+    """
+    A Permission Class that grants permission to users who's org_id matches the requested Domain's org_id.
+    """
+
+    def has_permission(self, request, view):
+        # Decode the identity header
+        try:
+            header_content = request.META.get('HTTP_X_RH_IDENTITY')
+            if header_content:
+                header_decoded_content = b64decode(header_content)
+            else:
+                return False
+        except Base64DecodeError:
+            return False
+        # Get the Org ID from the header
+        try:
+            header_value = json.loads(header_decoded_content)
+            json_path = jq.compile(".identity.internal.org_id")
```

Comment:
Can we move this to the module level so we will pre-compile the query just once?

Reply:
I'll move it. Thanks for the suggestion.